services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/S3TransferManager.java

```diff
@@ -281,8 +281,11 @@ default DirectoryUpload uploadDirectory(Consumer<UploadDirectoryRequest.Builder>
      * bucket will be downloaded.
      *
      * <p>
+     * The SDK will create the destination directory if it does not already exist.
```

Comment:
I think it will be almost guaranteed that there will be user-demand for this feature on something like download directory. It allows users to treat it like "sync" and it would allow it to behave symmetrical to upload directory (where S3 will not reject overwrites by default).

To be honest, I wish the default behavior were to overwrite-if-exists. I know that's not easily changeable, but I wonder if we should reconsider the default for new features like this.

Reply:
Yeah, I think we have to add overwrite-if-exists to support pause and resume before GA anyway. We can revisit the default behavior for download directory once we have that.